test/e2e/installplan_e2e_test.go

```diff
@@ -642,6 +642,203 @@ func TestCreateInstallPlanWithPreExistingCRDOwners(t *testing.T) {
+			require.Equal(t, 0, len(expectedSteps), "Actual resource steps do not match expected")
+
+			// new CRD changes
+			if tt.multipleVersions != nil {
+				for _, version := range tt.multipleVersions {
+					crdVersion := apiextensions.CustomResourceDefinitionVersion{
+						Name:    version,
+						Served:  true,
+						Storage: false,
+					}
+					mainCRD.Spec.Versions = append(mainCRD.Spec.Versions, crdVersion)
+				}
+				mainCRD.Spec.Version = ""
+				mainCRD.Spec.Versions[len(tt.multipleVersions)-1].Storage = true
+			} else {
+				mainCRD.Spec.Version = tt.version
+			}
+
+			var min float64 = 2
+			var max float64 = 256
+			if tt.changeSchema == true {
+				mainCRD.Spec.Validation = &apiextensions.CustomResourceValidation{
+					OpenAPIV3Schema: &apiextensions.JSONSchemaProps{
+						Properties: map[string]apiextensions.JSONSchemaProps{
+							"spec": {
+								Type:        "object",
+								Description: "Spec of a test object.",
+								Properties: map[string]apiextensions.JSONSchemaProps{
+									"scalar": {
+										Type:        "number",
+										Description: "Scalar value that should have a min and max.",
+										Minimum:     &min,
+										Maximum:     &max,
+									},
+								},
+							},
+						},
+					},
+				}
+			}
+
+			updateInternalCatalog(t, c, crc, mainCatalogSourceName, testNamespace, []apiextensions.CustomResourceDefinition{mainCRD}, []v1alpha1.ClusterServiceVersion{mainStableCSV, mainBetaCSV}, mainManifests)
+			// Attempt to get the catalog source before creating install plan(s)
+			_, err = fetchCatalogSource(t, crc, mainCatalogSourceName, testNamespace, catalogSourceRegistryPodSynced)
+			require.NoError(t, err)
+
+			// Update the subscription resource to point to the beta CSV
+			err = crc.OperatorsV1alpha1().Subscriptions(testNamespace).DeleteCollection(metav1.NewDeleteOptions(0), metav1.ListOptions{})
+			require.NoError(t, err)
+
+			// existing cleanup should remove this
+			createSubscriptionForCatalog(t, crc, testNamespace, subscriptionName, mainCatalogSourceName, mainPackageName, betaChannel, "", v1alpha1.ApprovalAutomatic)
+
+			subscription, err = fetchSubscription(t, crc, testNamespace, subscriptionName, subscriptionHasInstallPlanChecker)
+			require.NoError(t, err)
+			require.NotNil(t, subscription)
+
+			installPlanName = subscription.Status.Install.Name
+
+			// Wait for InstallPlan to be status: Complete or Failed before checking resource presence
+			fetchedInstallPlan, err = fetchInstallPlan(t, crc, installPlanName, buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseComplete, v1alpha1.InstallPlanPhaseFailed))
+			require.NoError(t, err)
+			fmt.Printf("Phase = %v", fetchedInstallPlan.Status.Phase)
+			t.Logf("Install plan %s fetched with status %s", fetchedInstallPlan.GetName(), fetchedInstallPlan.Status.Phase)
+
+			require.Equal(t, tt.expectedPhase, fetchedInstallPlan.Status.Phase)
+
+			// Fetch installplan again to check for unnecessary control loops
```

Comment:
I believe I need to remove this part.

Reply:
If it's not flaking you don't have to, but you can remove it if you want.